R/plotting-functions.R

```diff
@@ -259,9 +272,16 @@ plotSpectraOverlay <- function(x, xlab = "m/z", ylab = "intensity",
     if (frame.plot)
         box(...)
     title(main = main, xlab = xlab, ylab = ylab, ...)
+    if (length(labels)) {
+        if (is.function(labels)) {
+            labels <- labels(x)
+        }
+        if (length(labels) != length(x))
+            stop("Please provide a list of annotations of length equal to 'x'.")
+    } else {labels <- NULL}
```

Comment:
Do we need this? Isn't the `if (length(labels))` statement already enough?

Maybe, we should change the default to `label = list()` instead of the current `label = character()`.